Refactored semigraphics and configurable borders

benweidig opened the pull request:
Fixing horizontal/vertical in box.go, added comments to border runes for better visualization

rivo commented:
Thanks. Please give me some time to review. I will get back to you soon.

benweidig commented:
No hurry. Starting to use the changes in my own app so I can see if it works as intended, the demos looked good so far.

rivo commented on box.go:
```diff
@@ -236,27 +236,27 @@ func (b *Box) Draw(screen tcell.Screen) {
 		} else {
-			vertical = GraphicsHoriBar
-			horizontal = GraphicsVertBar
-			topLeft = GraphicsTopLeftCorner
```
I never noticed that I mixed up vertical and horizontal here. Thanks for catching this.

rivo commented on semigraphics.go:
```diff
@@ -0,0 +1,234 @@
+// SemigraphicJoints is a map for joining semigraphic (or otherwise) runes.
+// So far only light lines are supported but if you want to change the border
+// styling you need to provide the joints, too.
+var SemigraphicJoints = map[string]rune{
```
Have you tested all of these joints? Or did you generate them from the previous joint list? I want to be sure that these combinations are correct.

benweidig replied:
I "search&replace"-d them, and the table demo looked good. But I will check again and maybe add comments with the cimbined runes for a better understanding.

rivo commented on semigraphics.go:
```diff
@@ -0,0 +1,234 @@
+}
+
+// PrintJoinedBorder is the backwards-compatible wrapper for the new PrintJoinedSemigraphics.
+func PrintJoinedBorder(screen tcell.Screen, x, y int, ch rune, color tcell.Color) {
```
I think this can be removed. I know this will break compatibility but it wasn't a core function of the package anyway. (I'd be surprised if anybody uses it.)

rivo commented on util.go:
```diff
@@ -19,90 +19,29 @@ const (
-// Semigraphical runes.
+// Deprecated semigraphical runes.
+// You should use the runes provided by semigraphical.go or Borders directly.
 const (
```
Let's remove these, too.

benweidig commented:
Thx for the feedback, I'll change/update the PR accordingly in the next few days.

benweidig commented:
- SemigraphicJoints: Grid and Table demo are looking good, the comments were created with "Jump to Definition", so the Joints should all be correct. Also I've added a comment about the order of the runes, I haven't realized it before that it's important.

- Removed the PrintJoinedBorder and changed Grid accordingly

rivo commented:
Looks good. I merged it. Thanks again for the effort.
